api/src/test/java/org/openmrs/module/attachments/obs/TestHelper.java

```diff
@@ -234,4 +262,148 @@ public String getTestComplexObsFilePath() {
 		}
 		return obsList;
 	}
+	
+	/**
+	 * Boilerplate method to save a collection of complex obs.
+	 *
+	 * @param count The number of the complex obs to be saved.
+	 */
+	public List<Obs> saveComplexObsForEncounter(int count) throws IOException {
```

Comment:
Could you get rid for now of `saveComplexObs(int)` and `saveComplexObsForVisit(int)` and focus on `saveComplexObsForEncounter` for the sake of our first test.
Change its name and signature so that the target encounter is passed as a parameter, let make it something like that:
```java
public List<Obs> saveComplexObs(Encounter encounter, int count, int otherCount) {
  ...
}
```
Where `count` is the number of attachment-complex obs to save within the encounter and `otherCount` is the number of other complex obs to save within the encounter.

When you have that one up, we can look at the first unit test.